sentry-rails/lib/sentry/rails/action_cable.rb

```diff
@@ -2,20 +2,40 @@ module Sentry
   module Rails
     module ActionCableExtensions
       class ErrorHandler
-        def self.capture(env, transaction_name:, extra_context: nil, &block)
-          Sentry.with_scope do |scope|
-            scope.set_rack_env(env)
-            scope.set_extras(action_cable: extra_context) if extra_context
-            scope.set_transaction_name(transaction_name)
+        class << self
+          def capture(env, transaction_name:, extra_context: nil, &block)
+            Sentry.with_scope do |scope|
+              scope.set_rack_env(env)
+              scope.set_context("action_cable", extra_context) if extra_context
+              scope.set_transaction_name(transaction_name)
+              transaction = start_transaction(env, scope.transaction_name)
+              scope.set_span(transaction) if transaction
 
-            begin
-              block.call
-            rescue Exception => e # rubocop:disable Lint/RescueException
-              Sentry::Rails.capture_exception(e)
+              begin
+                block.call
+                finish_transaction(transaction, 200)
```

Comment:
This is more an open question. I'm not sure if assigning HTTP request semantics (here, status) to all websocket lifecycle events makes sense. 
I'm gonna check what python ASGI does for websockets and get back, but this can be merged and we can iterate later.

Reply:
ok nvm, python ASGI doesn't even set the status and for all practical purposes, websockets are currently treated similarly to HTTP requests so you can ignore this. :)